crates/collab/src/tests/editor_tests.rs

```diff
@@ -378,19 +378,10 @@ async fn test_collaborating_with_completion(cx_a: &mut TestAppContext, cx_b: &mu
         assert!(!buffer.completion_triggers().is_empty())
     });
 
-    // Type a completion trigger character as the guest.
-    editor_b.update_in(cx_b, |editor, window, cx| {
-        editor.change_selections(SelectionEffects::no_scroll(), window, cx, |s| {
-            s.select_ranges([MultiBufferOffset(13)..MultiBufferOffset(13)])
-        });
-        editor.handle_input(".", window, cx);
-    });
-    cx_b.focus(&editor_b);
-
-    // Receive a completion request as the host's language server.
-    // Return some completions from the host's language server.
-    cx_a.executor().start_waiting();
-    fake_language_server
+    // Set up the completion request handlers BEFORE typing the trigger character.
```

Comment:
I think the analysis is ok, yet most of the change is nonsense.

To me, seems that we need to use `initializer` field in `FakeLspAdapter` similar to 

https://github.com/zed-industries/zed/blob/e9073eceeba8e1a71e966582566179517591ec6b/crates/collab/src/tests/editor_tests.rs#L1953-L1985

instead of all this.

Reply:
Sorry for posting this sloppy comment, I will push see what can I apply on top and fix it further too.